itemadapter/adapter.py

```diff
@@ -361,17 +361,20 @@ def asdict(self) -> dict:
         """Return a dict object with the contents of the adapter. This works slightly different
         than calling `dict(adapter)`: it's applied recursively to nested items (if there are any).
         """
-        return {key: _asdict(value) for key, value in self.items()}
-
-
-def _asdict(obj: Any) -> Any:
-    """Helper for ItemAdapter.asdict()."""
-    if isinstance(obj, dict):
-        return {key: _asdict(value) for key, value in obj.items()}
-    if isinstance(obj, (list, set, tuple)):
-        return obj.__class__(_asdict(x) for x in obj)
-    if isinstance(obj, ItemAdapter):
-        return obj.asdict()
-    if ItemAdapter.is_item(obj):
-        return ItemAdapter(obj).asdict()
-    return obj
+        return {key: self._asdict(value) for key, value in self.items()}
+
+    @classmethod
+    def _asdict(cls, obj: Any) -> Any:
+        if isinstance(obj, dict):
+            return {key: cls._asdict(value) for key, value in obj.items()}
+        if isinstance(obj, (list, set, tuple)):
+            return obj.__class__(cls._asdict(x) for x in obj)
+        if isinstance(obj, cls):
+            return obj.asdict()
+        if cls.is_item(obj):
+            return cls(obj).asdict()
+        if isinstance(obj, ItemAdapter):
+            return obj.asdict()
+        if ItemAdapter.is_item(obj):
+            return ItemAdapter(obj).asdict()
```

Comment:
I wonder why keeping this would be needed.

Reply:
Probably a rare case, but we need to convert nested objects to dicts even if the ItemAdapter subclass itself does not handle the  https://github.com/scrapy/itemadapter/pull/68/commits/2732297ec28ee768b03d3ed27bf410d18649284f

Reply:
> we need to convert nested objects to dicts even if the ItemAdapter subclass itself does not handle them

Why do we need to do this? I'd expect that if a subclass e.g. removed support for dataclasses, that's on purpose, and user wants an exception to be raised. If user wants dataclasses support, it's easy to add it back explicitly.

Reply:
I think the issue could be that in the examples we don't show how to reuse the default set of adapter.

Reply:
@elacuesta I mean that in the examples you want to add an adapter to the existing adapter classes - this is essentially what was examples showing. But now it's a bit more complicated, because you need to use add something to the list from the base class. I'm fine with proceeding without these examples :)